Objects/stringlib/fastsearch.h

```diff
@@ -740,14 +740,31 @@ static inline Py_ssize_t
 STRINGLIB(count_char)(const STRINGLIB_CHAR *s, Py_ssize_t n,
                       const STRINGLIB_CHAR p0, Py_ssize_t maxcount)
 {
-    Py_ssize_t i, count = 0;
-    for (i = 0; i < n; i++) {
-        if (s[i] == p0) {
-            count++;
-            if (count == maxcount) {
-                return maxcount;
+    Py_ssize_t count = 0;
+    const STRINGLIB_CHAR *restrict cursor = s;
+    const STRINGLIB_CHAR *end_ptr = s + n;
+    const STRINGLIB_CHAR *unroll_end_ptr = end_ptr - 31;
+    /* By unrolling in chunks of 32, the compiler can auto vectorize, resulting
+       in much better performance. */
+    while (cursor < unroll_end_ptr) {
+        for(size_t i=0; i<32; i++) {
```

Comment:
```suggestion
        for(size_t i = 0; i < 32; i++) {
```

Let us keep the same style as before.